packages/cli/src/commands/deploy.ts

```diff
@@ -29,12 +53,24 @@ export default class Deploy extends Command {
       char: 'e',
       description: 'environment configuration to run',
     }),
+    skipRestoreDependencies: flags.boolean({
+      char: 's',
+      description: 'skips restoring dependencies after deployment'
+    })
   }
 
   public async run(): Promise<void> {
     const { flags } = this.parse(Deploy)
     if (initializeEnvironment(logger, flags.environment)) {
-      await runTasks(compileProjectAndLoadConfig(), deployToCloudProvider)
+      await runTasks(false, compileProjectAndLoadConfig(), deployToCloudProvider)
```

Comment:
Instead of passing the flag, you could pass here the `pruneDependencies` and `reinstallDependencies` functions already initialized with the flag option, or pass them conditionally only when the flag is true.